### PR TITLE
Coin-Cbc

### PR DESCRIPTION
**Pull request by scopatz**
This is an attempt to add the Coin-Cbc package.

### CODE REVIEW

**Comment by scopatz** [comment]
Does anyone understand why BLAS symbols wouldn't be found?  Is there something special that needs to be setup to get blas to work with conda-forge.  Note that this same recipe builds locally for me on Ubuntu 15.10.

**Comment by scopatz** [comment]
nm it looks like I just needed to include gcc as a build dep.

**Comment by whart222** [comment]
@scopatz I was planning on setting up this recipe with @tkralphs in early June.  He mentioned that he had a strategy for building COIN-OR packages that would be consistent for a bunch of different solvers: CLP, CBC, COUENNE, etc.  I'm including him to solicit his input.

**Comment by scopatz** [comment]
Hi @whart222 - OK great.  Unfortunately, I need this package ASAP in order to prepare for a Cyclus release.  I think a reasonable plan would be to proceed with this PR and add you (and maybe @tkralphs) as maintainers.  We can then take care of any improvements in the feedstock, when the pressure on me isn't so great :).  Does this sound like a plan?

**Comment by whart222** [comment]
I think that plan makes sense.

--Bill

**Comment by scopatz** [comment]
It looks like this builds! A review and merge would be much appreciated ;)

File: recipes/coincbc/build.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -e
+
+UNAME="$(uname)"
+export CFLAGS="-O3"
```
Comment (msarahan) [inline recipes/coincbc/build.sh]: Please never set CFLAGS and friends without grabbing their current value:

    export CFLAGS="${CFLAGS} -O3"

**Comment by msarahan** [comment]
LGTM other than clobbering CFLAGS and friends.  +1 to merge once that's fixed.

**Comment by scopatz** [comment]
Is Circle just totally broken right now? It seem to be failing to run simple commands like ls and `./configure`

**Comment by scopatz** [comment]
This worked yesterday

**Comment by ocefpaf** [comment]
Yep. See #613

**Comment by ocefpaf** [comment]
We need a twitter bot to broadcast this kind of failures.

**Comment by scopatz** [comment]
> We need a twitter bot to broadcast this kind of failures.

Yeah.  Well, as long as I am not significantly more crazy than I was yesterday :)

**Comment by jakirkham** [comment]
> We need a twitter bot to broadcast this kind of failures.

@pelson and I were talking about doing a status page ( https://github.com/conda-forge/conda-forge.github.io/issues/78 ). Basically the balls in my court. So, once I have time, I'm going to get started with it. We can figure out how to integrate with more stuff once it is live, but at least the info will live in one place that we can easily check.

**Comment by scopatz** [comment]
Hey @jakirkham, I think a status page is a great idea. If for no other reason than it saves your precious bandwidth :).  

In other news, this recipe finally builds. So thanks for working through these issues. Sorry I had to be mostly AFK this weekend.

**Comment by jjhelmus** [comment]
LGTM

From the reading I've done I think CBC can be build on Windows.  In fact the projects seems to have a [CI setup](https://github.com/coin-or/Cbc/blob/master/appveyor.yml) which does just that.  It does seem to be using MSYS2/MingW which might be a bit challenging. I don't have the time right now but in June I can try to get a Windows build working, maybe in the feedstock?

Either way really excited about getting this package into conda-forge.

**Comment by scopatz** [comment]
Thanks @jjhelmus! Would you like me to add you as a maintainer?

**Comment by jjhelmus** [comment]
Go for it.  Is this ready to merge once that is added?

**Comment by scopatz** [comment]
> Go for it. 

Thanks! Added.

> Is this ready to merge once that is added?

In my opinion, yes :)

**Comment by jjhelmus** [comment]
All green, merging

**Comment by jjhelmus** [comment]
Wait, I can't...

**Comment by scopatz** [comment]
> Wait, I can't...

no permissions?

**Comment by jjhelmus** [comment]
> no permissions?

Yup.

**Comment by scopatz** [comment]
Hopefully @ocefpaf or @msarahan or someone can merge it in :)

**Comment by ocefpaf** [comment]
@jjhelmus do you want merge rights here BTW? We are looking for help :wink:

**Comment by scopatz** [comment]
Thanks @ocefpaf!

**Comment by tkralphs** [comment]
Hi all,

Sorry for not weighing in earlier! I was pretty flat out and just didn't have the bandwidth to look at this in the time span you needed it. 

As @whart222 pointed out, I have indeed put together a [general build script](https://github.com/coin-or-tools/BuildTools/blob/stable/0.8/get.dependencies.sh) that is being used for builds on [Travis](https://travis-ci.org/coin-or) and Appveyor for all projects. Right now, it pulls source from either git or svn, but I could add the option to pull a tarball. It was written with the idea that one would pull the source for just the project itself first, and then pull the sources for the dependencies separately, since on Github, cloning a repo doesn't pull in dependencies automagically, as on SVN. You can see how it's done by looking at the [travis.yml](https://projects.coin-or.org/Cbc/browser/trunk/.travis.yml) for Cbc.

I'll talk this over with @whart222 and try to come up with an overall strategy for constructing recipes for all projects. I have a feeling that it's going to be better not to do monolithic builds, but to have recipes depend on one another (if this is allowed in Conda). This is how we're doing it in the [homebrew recipes](https://github.com/coin-or-tools/homebrew-coinor)  being constructed for OS X.

**Pull request merged**